### PR TITLE
Rtrace

### LINKED ISSUE #55: [Feature RFC] Implement reference count reporting
Please use this issue to discuss refcount measurements for testing.

### LINKED ISSUE #72: Easy way to compare arrays?
I tried 

```ts
const array: f64[] = [20.0,30.0,0,0,0,0]
expect<f64[]>(array).toBe([20.0,30.0,0,0,0,0])
```

which may not even be the same types of array, but I realized that this compares them by reference (obviously it would fail).

Is there an API to easily compare arrays? Or should I loop on the indexes and use expect on each value for now?

### LINKED ISSUE #80: How to add console.log to test source code
I'd like to add console.logs inside my code to see what's going on.

In `as-pect.config.js`, I've got

```js
	imports: {
		console: {
			log(msg) {
				console.log(msg)
			},
		},
	},
```

In a `globals.d.ts` file I have

```ts
declare const console: {
	log(a: any): void
}
```

In my AS source I have things like `console.log(whatever)` and my IDE doesn't have any issue with it.

But when I run tests it says

```
ERROR TS2304: Cannot find name 'console'.

                console.log('HELLO --------------------- ')
   ~~~~~~~
```

My `globals.d.ts` is listed in `tsconfig.json`, and it works fine with `asc`.

Is there something I have to do for it to work specifically in `asp`?

### LINKED ISSUE #82: all tests pass, but summary shows a failure
I've ran into an interesting case. Here's my output:

```
✔ src/as/glas/core/Layers.spec.ts Pass: 5 / 5 Todo: 0 Time: 4.59ms
✔ src/as/glas2/Math/BezierCurve.spec.ts Pass: 1 / 1 Todo: 0 Time: 0.337ms
✔ src/as/glas2/Math/MathTemp.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.047ms
✔ src/as/glas2/Math/Matrix.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.068ms
✔ src/as/glas2/Math/Quaternion.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.059ms
✔ src/as/glas2/Math/Scalar.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.045ms
✔ src/as/glas2/Math/Size.spec.ts Pass: 12 / 12 Todo: 0 Time: 2.72ms
✔ src/as/glas2/Math/Vector2.spec.ts Pass: 35 / 35 Todo: 0 Time: 8.232ms
✔ src/as/glas2/Math/Vector3.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.044ms
✔ src/as/glas2/Math/Vector4.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.05ms
✔ src/as/glas2/Misc/ArrayTools.spec.ts Pass: 0 / 0 Todo: 0 Time: 0.048ms
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  [Result]: ✖ FAIL
   [Files]: 12 total
  [Groups]: 55 count, 55 pass
   [Tests]: 53 pass, 3 fail, 56 total
    [Time]: 1840.689ms
```

Shows that all tests passed, but the numbers aren't matching in the summary. It does correctly say `53` tests passed, but additionally it says 3 tests failed.

### PR DESCRIPTION
**Pull request by jtenner**
Implement the `RTrace` interface.

- [x] Add allocation errors to group warnings
- [x] Add all relevant properties to RTrace to `TestContext`
- [x] Add all relevant properties to RTrace to `TestResult` 
- [x] Add all relevant properties to RTrace to `TestGroup`
- [x] Use custom rtrace interface object instead of default (reset allocation and deallocation counts)
- [x] Implement rtrace counts on DefaultTestReporter in non-zero instances
- [x] Jest code coverage 100%
- [x] New RTrace API
  - [x] RTrace.end(label: i32);
  - [x] RTrace.frees();
  - [x] RTrace.groupAllocations();
  - [x] RTrace.groupDecrements();
  - [x] RTrace.groupFrees();
  - [x] RTrace.groupIncrements();
  - [x] RTrace.increments();
  - [x] RTrace.start(label: i32);
  - [x] RTrace.testAllocations();
  - [x] RTrace.testDecrements();
  - [x] RTrace.testFrees();
  - [x] RTrace.testIncrements();
- [x] Documentation
- [x] Fix SummaryTestReporter false pass indication
- [x] Fix `--reporter` flag with `querystring.parse()`
- [x] Add Warning output when using `--reporter` flag and falls back on `DefaultTestReporter`
- [x] Add parameter to `SummaryTestReporter` called `loggingEnabled`

Fixes #55 
Fixes #82
Fixes #80 
Fixes #72

### CODE REVIEW

**Comment by jtenner** [comment]
Only one more line of code to cover reasonably.

**Comment by jtenner** [comment]
This pull request is now ready for code review @willemneal. You don't have to approve this pull request just yet, just please review the code changes and ask questions. Thanks!

**Comment by jtenner** [comment]
@willemneal This pull request is ready for review. It will result in a `1.1.0` release.

File: src/cli/run.ts [hunk 1]
```diff
@@ -163,6 +180,11 @@ export function run(yargs: IYargs, compilerArgs: string[]): void {
 
   const folderMap = new Map<string, string[]>();
   const fileMap = new Map<string, string>();
+  console.log(chalk`{bgWhite.black [Log]} Effective command line arguments:`);
```
Comment (willemneal) [inline src/cli/run.ts]: Nice touch.

**Pull request merged**